jepsen/src/jepsen/control.clj

```diff
@@ -360,6 +360,35 @@
 
 (def ssh "A remote that does things via clj-ssh." (SSHRemote. nil))
 
+(defn- undo-command-changes
+  "The `wrap-sudo` and `wrap-cd` functions adds stuff in front of commands which
+  makes docker crap itself, this function undoes that."
+  [cmd]
+  (-> cmd
+      (str/replace-first "sudo -S -u root" "")
+      (str/replace-first "cd /; " "")))
```

Comment:
Ooh, uhhh, this kind of string munging looks fraught with peril. Maybe it would make more sense to move command wrappers into the Remote protocol?

Reply:
I was wondering if I can make `wrap-cd` and `wrap-sudo` not add those strings by using `cd nil` and `sudo nil`, but that's not going to be possible, because that happens before `execute!`, right?

Reply:
That sounds right, yeah. I'm confused about why this is even a thing we have to do, cuz, like... the sudo and directory management stuff is there for a reason. Undoing it sounds like something that could backfire in terrible ways.

Reply:
Some docker images do not seem to have `/` in their filesystem, so `cd /` fails. Likewise with `sudo` most docker images only have one user, `root`, sometimes `sudo` isn't even installed.

Reply:
<sigh> Yeah, that's gonna cause *all kinds* of breakage for tests that set up specific users.

Reply:
(tbh I think the answer here is probably "don't use docker"; it's... almost certainly less work to set up actual VMs or real containers instead of trying to work around behavior like this)

Reply:
I must have been doing something wrong in the early development of this PR. It doesn't make sense that `cd /` would work over `ssh` but not over `docker exec`, I just tried removing `undo-command-changes` completely and it works. Updated PR.